src/Query/Builder.php

```diff
@@ -979,21 +987,26 @@ public function runPaginationCountQuery($columns = ['*'])
     /**
      * Perform an update query.
      *
-     * @param  array $query
-     *
      * @return int
      */
-    protected function performUpdate($query, array $options = [])
+    protected function performUpdate(array $update, array $options = [])
     {
         // Update multiple items by default.
         if (! array_key_exists('multiple', $options)) {
             $options['multiple'] = true;
         }
 
+        // Since "id" is an alias for "_id", we prevent updating it
```

Comment:
TODO: support alias for nested "id" field.
what if `$update` is a pipeline? That becomes too complex.